Sources/OktaDirectAuth/Internal/Authentication Factors/SecondaryFactor.swift

```diff
@@ -28,41 +28,73 @@ extension DirectAuthenticationFlow.SecondaryFactor: AuthenticationFactor {
             return TokenStepHandler(flow: flow, request: request)
         case .oob(channel: let channel):
             return try OOBStepHandler(flow: flow,
                                       openIdConfiguration: openIdConfiguration,
+                                      currentStatus: currentStatus,
                                       loginHint: loginHint,
-                                      mfaToken: currentStatus?.mfaToken,
                                       channel: channel,
                                       factor: factor,
                                       bindingContext: bindingContext)
+        case .webAuthn:
+            let mfaContext = currentStatus?.mfaContext
+            let request = try WebAuthnChallengeRequest(openIdConfiguration: openIdConfiguration,
+                                                       clientConfiguration: flow.client.configuration,
+                                                       loginHint: loginHint,
+                                                       mfaToken: mfaContext?.mfaToken)
+            return ChallengeStepHandler(flow: flow, request: request) {
+                .webAuthn(.init(request: $0,
+                                mfaContext: mfaContext))
+            }
+        case .webAuthnAssertion(let response):
+            let request = TokenRequest(openIdConfiguration: openIdConfiguration,
+                                       clientConfiguration: flow.client.configuration,
+                                       currentStatus: currentStatus,
+                                       loginHint: loginHint,
+                                       factor: factor,
+                                       parameters: response,
+                                       grantTypesSupported: flow.supportedGrantTypes)
+            return TokenStepHandler(flow: flow, request: request)
         }
     }
     
-    var tokenParameters: [String: Any]? {
+    func tokenParameters(currentStatus: DirectAuthenticationFlow.Status?) -> [String: String] {
+        var result: [String: String] = [
+            "grant_type": grantType(currentStatus: currentStatus).rawValue,
+        ]
+        
+        if let context = currentStatus?.mfaContext {
+            result["mfa_token"] = context.mfaToken
+        }
+
         switch self {
         case .otp(code: let code):
-            return [
-                "grant_type": grantType.rawValue,
-                "otp": code
-            ]
-        case .oob:
-            return nil
+            result["otp"] = code
+        case .webAuthnAssertion(_): break
+        case .oob(channel: _): break
+        case .webAuthn: break
         }
 
+        return result
     }
 
-    var grantType: GrantType {
+    func grantType(currentStatus: DirectAuthenticationFlow.Status?) -> GrantType {
         switch self {
         case .otp:
             return .otpMFA
         case .oob:
             return .oobMFA
```

Comment:
can you add the check for mfaToken here too to determine the right grant type to fix the issue with push factor number challenge?

Reply:
I'll do that as a separate PR, since I didn't want to bloat this one any more than it already has.